cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1480,7 +1486,51 @@ func maybeMountSave(disk disks.Disk, rootdir string, encrypted bool, mountOpts *
 	return true, nil
 }
 
+func populateSysroot(mst *initramfsMountsState) error {
+	mntOpts := &systemdMountOptions{
+		Bind:      true,
+		Ephemeral: true,
+	}
+	logger.Noticef("Creating dir %s", boot.InitramfsSysroot)
+	err := os.MkdirAll(boot.InitramfsSysroot, 0755)
+	if err != nil {
+		return err
+	}
+
+	// Classic
+	if mst.isClassic {
+		logger.Noticef("Mounting classic bits")
+		// /run/mnt/data in /sysroot
+		if err := doSystemdMount(boot.InitramfsDataDir, boot.InitramfsSysroot, mntOpts); err != nil {
+			return err
+		}
+	} else {
+		logger.Noticef("Mounting UC bits ephemeral")
+		// /run/mnt/base in /sysroot
+		if err := doSystemdMount(
+			filepath.Join(boot.InitramfsRunMntDir, snapTypeToMountDir[snap.TypeBase]),
+			boot.InitramfsSysroot, mntOpts); err != nil {
+			return err
+		}
+		// /run/mnt/data in /sysroot/writable
+		if err := doSystemdMount(boot.InitramfsDataDir,
+			boot.InitramfsSysrootWritable, mntOpts); err != nil {
+			return err
+		}
+
+		// We need to enable and start. TODO find out why.
+		sysd := systemd.New(systemd.SystemMode, nil)
+		err = sysd.EnableAndStartNoBlock([]string{"populate-writable.service"})
```

Comment:
I think we could do that in `initrd-root-fs.target`. Just make make a small service unit that looks at the root file system, then detect if it UC and add `populate-writable.service` to `initrd-fs.target`.

`initrd-parse-etc.service` will do a `daemon-reload` and retrigger `initrd-fs.target`. This means that if you enabled it, it then start.

Reply:
Right, but the problem is actually the detection of whether this is UC or classic. We do that by looking at the model assertion, and that is more easily done inside `snap-bootstrap`, after you have mounted the boot partition - also we verify there the model, which cannot be done easily elsewhere.

If eventually we decide to use different initramfs for UC/classic that would change things.